dakara_server/playlist/views.py

```diff
@@ -318,10 +265,167 @@ def perform_update(self, serializer):
         karaoke = serializer.instance
 
         if karaoke.status == models.Karaoke.STOP:
-            player = models.Player.get_or_create()
-            player.reset()
+            # request the player to be idle
+            broadcast_to_channel('playlist.device', 'send_idle')
+
+            # clear player
+            player = models.Player()
             player.save()
+
+            # empty the playlist
             models.PlaylistEntry.objects.all().delete()
 
+            # empty the player errors
+            models.PlayerError.objects.all().delete()
+
+        elif karaoke.status == models.Karaoke.PLAY:
+            player = models.Player.get_or_create()
+
+            # request the player to play the next song if idle
+            if player.playlist_entry is None:
+                broadcast_to_channel(
+                    'playlist.device', 'send_playlist_entry', data={
+                         'playlist_entry': models.PlaylistEntry.get_next()
+                     }
+                )
```

Comment:
The exception is expected in this case, as I want to avoid to send a `None` playlist entry.

I am a bit surprised that `send_playlist_entry` is broadcasted directly, since `handle_next` is the preferred action, which checks if there is something to play beforehand. This is the simplest solution, but it will resend the `send_idle` action to the player if there is nothing to play. Otherwise, we can reproduce some bits of the logic of `handle_next` here.

Reply:
I just pushed a test, and a fix for this issue.

I reproduced some of the handle next logic like in your code above
(actually I did the exact same code)

Reply:
But I'm not very happy with the test I made, I simply assert broadcast to channel is not called.
Maybe I should assert it is not called with a None playlist entry

Reply:
It is stronger to assert that `broadcast_to_channel` is not called at all than not called with a specific argument. Moreover, the method is not supposed to be called.

Reply:
Ok, let's keep it that way then.